src/MasterSupervisor.php

```diff
@@ -56,15 +56,23 @@ class MasterSupervisor implements Pausable, Restartable, Terminable
      */
     public static $nameResolver;
 
+    /**
+     * The environment that is used to provision this master supervisor.
+     *
+     * @var string|null
+     */
+    public $environment;
+
     /**
      * Create a new master supervisor instance.
      *
      * @return void
      */
-    public function __construct()
+    public function __construct(?string $environment = null)
```

Comment:
The `?` is not needed here because the argument is already nullable.

```suggestion
    public function __construct(string $environment = null)
```